src/earthengine/ee_worker.js

```diff
@@ -164,38 +186,49 @@ class EarthEngineWorker {
 
     // Returns raster tile url for a classified image
     getTileUrl() {
-        const { format, data } = this.options
+        const { datasetId, format, data, filter, style } = this.options
 
-        return new Promise(resolve => {
-            if (format === 'FeatureCollection') {
-                const { datasetId } = this.options
+        return new Promise((resolve, reject) => {
+            switch (format) {
+                case FEATURE_COLLECTION:
+                    let dataset = ee.FeatureCollection(datasetId)
 
-                let dataset = ee
-                    .FeatureCollection(datasetId)
-                    .draw(FEATURE_STYLE)
+                    dataset = applyFilter(dataset, filter).draw({
+                        ...DEFAULT_FEATURE_STYLE,
+                        ...style,
+                    })
 
-                if (data) {
-                    dataset = dataset.clipToCollection(
-                        this.getFeatureCollection()
-                    )
-                }
+                    if (data) {
+                        dataset = dataset.clipToCollection(
+                            this.getFeatureCollection()
+                        )
+                    }
 
-                dataset.getMap(null, response => resolve(response.urlFormat))
-            } else {
-                let { eeImage, params } = getClassifiedImage(
-                    this.getImage(),
-                    this.options
-                )
+                    dataset.getMap(null, response =>
+                        resolve(response.urlFormat)
+                    )
 
-                if (data) {
-                    eeImage = eeImage.clipToCollection(
-                        this.getFeatureCollection()
+                    break
+                case IMAGE:
+                case IMAGE_COLLECTION:
+                    let { eeImage, params } = getClassifiedImage(
+                        this.getImage(),
+                        this.options
                     )
-                }
 
-                eeImage
-                    .visualize(params)
-                    .getMap(null, response => resolve(response.urlFormat))
+                    if (data) {
+                        eeImage = eeImage.clipToCollection(
+                            this.getFeatureCollection()
+                        )
+                    }
+
+                    eeImage
+                        .visualize(params)
+                        .getMap(null, response => resolve(response.urlFormat))
+
+                    break
+                default:
+                    reject(new Error('Unknown format'))
```

Comment:
This error message should be a little more informative since it gets displayed to the user. For example, "The data format for Earth Engine layer [x] was not recognized"